### PR TITLE
fix: After editing the pipeline configuration, the pipeline disappears

### PR DESCRIPTION
**Pull request by weili520**
Signed-off-by: TheYoungManLi <cjl@kubesphere.io>

### What type of PR is this?
/kind bug

### Which issue(s) this PR fixes:
Fixes ##2810

### Special notes for reviewers:

https://user-images.githubusercontent.com/33231138/145422191-ce5aeebd-ffe6-4551-b700-8ed381ca343e.mov

### Does this PR introduced a user-facing change?
```release-note
NONE
```

### CODE REVIEW

File: src/pages/devops/containers/Pipelines/Detail/Layout/pipeline.jsx [hunk 1]
```diff
@@ -143,8 +144,9 @@ export default class PipelineDetailLayout extends React.Component {
         type: 'control',
         text: t('EDIT_SETTINGS'),
         action: 'edit',
-        onClick: () => {
+        onClick: async () => {
           this.getPipeLineConfig()
+          await this.store.getJenkinsFile(params)
```
[the remark this answers is not in the document]
Comment (JohnNiang) [reply src/pages/devops/containers/Pipelines/Detail/Layout/pipeline.jsx:149]: Hi @weili520 @harrisonliu5 , what do you reckon about this worry?

[the remark this answers is not in the document]
Comment (weili520) [reply src/pages/devops/containers/Pipelines/Detail/Layout/pipeline.jsx:149]: Yes, you are right, This problem caused by we are not getting the latest pipeline config information before updating it.

**Review by JohnNiang — APPROVED** [review]
/lgtm
@weili520 Thanks for your effort on this problem. I have confirmed that it works very well.

image

**Pull request merged**